WIP: Update manual setup process

bkmgit opened the pull request:
Add more details on manual setup process.

Replace occurrences of yum with dnf.

pypingou committed:
Encode the image title to UTF-8 for non-ascii characters

jeremycline committed:
Make the CSRF token timeout configurable

Previously, the ``WTF_CSRF_TIME_LIMIT`` setting was not in the default
configuration, nor was it documented. It also only worked in
flask-wtf-0.10.1+, which is not available in EPEL7. This makes it both
work in EPEL7's version (0.8) and documents the setting.

fixes #78

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Add a Vagrant environment

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Clarify the CSRF documentation in default_config

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Add travis support and fix broken tests

Depending on how the tests were run, the test for the pictures endpoint
would fail because the path to the pictures folder would be incorrectly
calculated since it could be relative. This ensures the path is always
absolute.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Switch back from unittest2 to unittest

I very briefly fiddled with running the tests on 2.6, but decided it
wasn't worth the effort. However, I neglected to add the modified
.travis.yml file when I updated #108 so this fixes that.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

pypingou committed:
Fix import error due to recent changes in flask

pypingou committed:
Fix the minimal flask-wtf version not working as expected

pypingou committed:
Small style changes

pypingou committed:
Drop two un-needed instantiators.

pypingou committed:
Add an submission_date_end field to the Election

This will allow to plan some time between the end of the submission
phase and the start of the voting phase.

Add an alembic migration script to add the submission_date_end field

pypingou committed:
Include the coverage as the dependency to run the tests

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Port nuancier to python3 while remaining compatible with python2

Included the tests

Make the tests against 3.4 and 3.5 not allowed to fail anymore

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Improve the MANIFEST.in to exclude some un-necessary files

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Release 0.11.0

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

00willo committed:
Gets 'vagrant up' to work from fresh git clone

Commit:
Old style exceptions --> new style in Python 3

Python 3 treats old style exceptions as syntax errors but new style exceptions work as expected in both Python 2 and Python 3.

pypingou committed:
Clean up a little bit deprecated code

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Add support for tox to run the tests

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Add .cico.pipeline file

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Pass onto tox the HOME variable

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

xsuchy committed:
check if picture is landscape oriented

pypingou committed:
Adjust a little bit travis and tox to test more

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Run tox in verbose mode

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Make travis call nose in verbose mode

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Make nuancier's code forward compatible with new version of flask-wtf

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Make it explicit that 'votes DESC' is a text representation of an SQL statement

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Make the tests pass with newer version that return 308 instead of 301

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

pypingou committed:
Ensure the order remains consistent

Signed-off-by: Pierre-Yves Chibon <pingou@pingoured.fr>

JochemKuijpers committed:
Add a title attribute to status image icons matching their alt attribute.

Update README.rst

Remove broken link to image that was at

.. image:: http://shieldkins.elrod.me/fedora/nuancier

Update README.rst

Add git to things to install

Update development.rst

Update development.rst

Change yum to dnf

Update deployment.rst

Change yum to dnf, however not clear which system wide repositories nuancier is in.

Update README.rst

Update manual install directions

Update README.rst

improve formatting

Update README.rst

formatting fixes, update information on requirements

Update README.rst

Co-Authored-By: Clement Verna <cverna@users.noreply.github.com>

Update development.rst

Update development.rst

Change yum to dnf

Update deployment.rst

Change yum to dnf, however not clear which system wide repositories nuancier is in.

Update Vagrantfile.example

Use Fedora 30 image instead of Fedora 25

Update main.yml

Use Python3
Update to use loop syntax

Update Vagrantfile.example

Remove , sshfs_opts_append: "-o nonempty" since the nonempty option is no longer available.

Update Vagrantfile.example

Use Python3 in Vagrant example file

cverna commented on README.rst:
```diff
@@ -76,22 +76,41 @@ All done! The VM contains all the required dependencies pre-installed.
 
+or to run the server externally, first open relevant port in firewall::
+
+    sudo firewall-cmd --add-port 3000/tcp --permanent
```
```suggestion
    sudo firewall-cmd --add-port 5000/tcp --permanent
```

cverna replied:
You could also forward the port in the Vagrantfile that way you can connect to the application from your host. See https://github.com/fedora-infra/bodhi/blob/develop/Vagrantfile#L53 as an example.

bkmgit replied:
Thanks. This looks good. Will also update [example vagrant file](https://github.com/fedora-infra/nuancier/blob/master/Vagrantfile.example) to use a more recent Fedora image and to use Python3

cverna reviewed (commented):
LGTM, just one small comment

cverna commented:
Looks good :-), It would be good to squash the 8 commits into a single one so that you keep a cleaner history. Here is a good guide on how to do that --> https://stackabuse.com/git-squash-multiple-commits-in-to-one-commit/

bkmgit commented:
Ok - will squash commits. For whatever reason, the build on Python 3.4 seems to fail - though no code changes have been made.

cverna commented:
> Ok - will squash commits. For whatever reason, the build on Python 3.4 seems to fail - though no code changes have been made.

It is ok to ignore, the failure is due to a version conflict for one of the dependency. TBH we should not test on Python 3.4, Python 3.5 and Python 3.6 and target directly Python 3.7

bkmgit commented:
Added separate pull request, #125 with changed tests.

cverna commented:
> Added separate pull request, #125 with changed tests.

Merged :-)

bkmgit committed:
Update 7db0e24a2a85_add_submission_end_date.py

Change print statement to be compatible with Python 3

sebwoj commented on devel/ansible/roles/nuancier-dev/tasks/main.yml:
```diff
@@ -1,16 +1,20 @@
 - name: Install helpful development packages
-  dnf: name={{ item }} state=present
-  with_items:
+  dnf: 
```
Lets remove extra space on the end.

bkmgit replied:
ok

sebwoj commented on devel/ansible/roles/nuancier-dev/tasks/main.yml:
```diff
@@ -1,16 +1,20 @@
-  dnf: name={{ item }} state=present
-  with_items:
+  dnf: 
+    name: "{{ item }}" 
```
Lets remove extra space on the end.

bkmgit replied:
ok

sebwoj commented on devel/ansible/roles/nuancier-dev/tasks/main.yml:
```diff
@@ -1,16 +1,20 @@
-  dnf: name={{ item }} state=present
-  with_items:
+  dnf: 
+    name: "{{ item }}" 
```
Lets remove extra space on the end.

bkmgit replied:
ok

sebwoj commented on Vagrantfile.example:
```diff
@@ -9,11 +9,11 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box_url = "https://download.fedoraproject.org/pub/fedora/linux/releases"\
-                     "/25/CloudImages/x86_64/images/Fedora-Cloud-Base-Vagrant-25-1"\
-                     ".3.x86_64.vagrant-libvirt.box"
+                     "/30/Cloud/x86_64/images/Fedora-Cloud-Base-Vagrant-30-1"\
```
Why not using the latest release(f31) here?

bkmgit replied:
Was not available when tested this, but will update

bkmgit replied:
Updated.

Commit:
update werkzeug library to invoke required workzeug.utils

update test requirements

remove Python2 from rpm dependencies

change dependency from rpm-python to python3-rpm

update travis ci file to use only python 3.7

Update Vagrantfile.example

Use Fedora 32 image instead of Fedora 25

change dependency from rpm-python to python3-rpm

update test requirements

remove Python2 from rpm dependencies

change dependency from rpm-python to python3-rpm

update travis ci file to use only python 3.7

Update Vagrantfile.example

Use Fedora 32 image instead of Fedora 25

Update README.rst

Remove broken link to image that was at

.. image:: http://shieldkins.elrod.me/fedora/nuancier

Update README.rst

Add git to things to install

Update development.rst

Update development.rst

Change yum to dnf

Update deployment.rst

Change yum to dnf, however not clear which system wide repositories nuancier is in.

Update README.rst

Update manual install directions

Update README.rst

improve formatting

Update README.rst

formatting fixes, update information on requirements

Update README.rst

Co-Authored-By: Clement Verna <cverna@users.noreply.github.com>

Update development.rst

Update development.rst

Change yum to dnf

Update deployment.rst

Change yum to dnf, however not clear which system wide repositories nuancier is in.

Update Vagrantfile.example

Remove , sshfs_opts_append: "-o nonempty" since the nonempty option is no longer available.

Fedora 32 no longer supports rpm-python, updated to python3-rpm which seems to cover everything rpm-python was doing

jontrossbach commented:
@bkmgit I just made a PR to your fork. Perhaps, the changes I made should be included in this pull.

jontrossbach reviewed (changes requested):
please read my comment in the conversation of this PR

jontrossbach commented:
@bkmgit I think the changes you made on top of my changes broke the dev-env set up. It appears that Fedora 32 no longer has an rpm named rpm-python and it is breaking the Ansible playbook that is run from the Vagrantfile. Could you try setting up a dev-env on your machine an telling me if you run into the same issue?

jontrossbach reviewed (approved):
As far as I can tell, this branch is ready to be pulled in now.

pypingou commented:
The commit history for this PR needs to be cleaned up a little bit before we merge it. The PR needs to be rebased and the commits that belong together be squashed.

pypingou commented on README.rst:
```diff
@@ -39,7 +37,7 @@ order to increase the loading speed of the pages.
 The dependency list is therefore:
 
-- `python`_ (2.5 minimum)
+- `python`_ (3.7 minimum)
```
travis.yml seems to still be testing 3.6 ;-)

bkmgit replied:
Thanks. Will remove.

pypingou commented on README.rst:
```diff
@@ -78,22 +76,41 @@ All done! The VM contains all the required dependencies pre-installed.
+or to run the server externally, first open relevant port in firewall::
+
+    sudo firewall-cmd --add-port 5000/tcp --permanent
+    sudo firewall-cmd --reload 
```
:-1: let's not advice people to poke holes into their firewall, ``runserver.py`` has a ``--port``argument iirc and if there is a port conflict with apache, when nuancier should be run via wsgi or apache configured to do port redirection, but I don't think this belongs to the documentation on how to get nuancier running for **development**.

bkmgit replied:
Thanks.

pypingou commented:
So 2 comments to address and once this is done and the history cleaned up a little, I think this can get in :)

bkmgit commented:
@pypingou Squashed commits and addressed the 2 comments. Hopefully ok to merge this pull request. 

Started on using OIDC in another branch: 
https://github.com/bkmgit/nuancier/tree/newauth 
changes in 
https://github.com/bkmgit/nuancier/blob/newauth/nuancier/__init__.py

pypingou commented:
@bkmgit I'm seeing 33 commits in this PR and it looks like the branch is now conflicting, so I think there is still some work to do ;-)

pypingou commented on .travis.yml:
```diff
@@ -1,10 +1,7 @@
- - "3.4"
- - "3.5"
- - "3.6"
+ - "3.7"
```
Why removing the old py3 version? Are we making any changes here that bump the minimum version of python required? (Cf also the minimum version set in the README below).

bkmgit replied:
@pypingou Thanks for the feedback. Using python 3.7 as per suggestion by CVerna.
